src/web/components/Account/CreateAccountForm.tsx

```diff
@@ -0,0 +1,146 @@
+import { useState } from 'react';
+import { FormProvider, useForm } from 'react-hook-form';
+
+import { ParticipantTypeDTO } from '../../../api/entities/ParticipantType';
+import { UserRole } from '../../../api/entities/User';
+import { CreateParticipantForm } from '../../services/participant';
+import { Dialog } from '../Core/Dialog';
+import { withoutRef } from '../Core/Form';
+import { TermsAndConditions } from '../Core/TermsAndConditions';
+import { CheckboxInput } from '../Input/CheckboxInput';
+import { FormError, RootFormErrors, setGlobalErrors } from '../Input/FormError';
+import { SelectInput } from '../Input/SelectInput';
+import { FormStyledCheckbox } from '../Input/StyledCheckbox';
+import { TextInput } from '../Input/TextInput';
+
+export type CreateAccountFormProps = {
+  resolvedParticipantTypes: ParticipantTypeDTO[];
+  onSubmit: (data: CreateParticipantForm) => Promise<string[] | void>;
+};
+export function CreateAccountForm({ resolvedParticipantTypes, onSubmit }: CreateAccountFormProps) {
+  const [showTermsDialog, setShowTermsDialog] = useState(false);
+  const formMethods = useForm<CreateParticipantForm>({
```

Comment:
I had to switch to the hook-based `useForm` method for a few things, including `watch` (needed to disable the submit button until the terms have been read - see Figma).